### PR TITLE
contrib sklearnSVD-TfSVD files

### PR DESCRIPTION
**Pull request by leoninekev**

### CODE REVIEW

**Review by dhavala — APPROVED** [review]

**Review by dhavala — APPROVED** [review]

**Review by dhavala — APPROVED** [review]

**Review by dhavala — CHANGES_REQUESTED** [review]
all the logic went into adapter at sklearndecompose. creating new adapters should be done very rarely. in this case, creating an adapter for sklearn SVD seems unnecessary. sklearn-svd is actually based on sklearn base model class. try not creating sklearndecompose class

**Review by dhavala — CHANGES_REQUESTED** [review]
this new class in adapters is probably unnecessary. try implementing without this adapter, and instead use keras-regressor. you can create a new class model in architecture, just like we have GeneralizedLinearModel for all GLMs (logisticsic, linear, SVM etc). All core model logic went into this adapter.

**Review by dhavala — APPROVED** [review]

**Review by dhavala — APPROVED** [review]

**Review by dhavala — APPROVED** [review]

**Review by dhavala — APPROVED** [review]

**Review by dhavala — APPROVED** [review]

**Review by dhavala — APPROVED** [review]

**Review by dhavala — APPROVED** [review]

**Review by dhavala — APPROVED** [review]

**Comment by leoninekev** [comment]
sir, please take a look.

File: src/mlsquare/adapters/sklearn.py [hunk 1]
```diff
@@ -222,14 +244,17 @@ def predict(self, X):
...
+            raise AttributeError("'SklearnKerasRegressor' object has no attribute 'predict'")
         pred = self.final_model.predict(X)
         return pred
 
     def save(self, filename=None):
         if filename == None:
             raise ValueError(
                 'Name Error: to save the model you need to specify the filename')
-
+        if isinstance(self.proxy_model, (sklearn.DimensionalityReductionModel)):
```
Comment (dhavala) [inline src/mlsquare/adapters/sklearn.py:256]: why are still you checking for these kind of isinstance of kind of checks? try avoiding.

File: src/mlsquare/adapters/sklearn.py [hunk 2]
```diff
@@ -2,10 +2,12 @@
 # -*- coding: utf-8 -*-
 from ..optmizers import get_best_model
 from ..utils.functions import _parse_params
+from ..architectures import sklearn
 import pickle
 import onnxmltools
 import numpy as np
 
+
```
Comment (dhavala) [inline src/mlsquare/adapters/sklearn.py:10]: entire SklearnKerasRegressor is filled with isinstance checks. This should be avoided.

File: src/mlsquare/architectures/sklearn.py [hunk 3]
```diff
@@ -85,6 +87,83 @@ def update_params(self, params):
     def adapter(self):
         return self._adapter
 
+class DimensionalityReductionModel:
```
Comment (dhavala) [inline src/mlsquare/architectures/sklearn.py:90]: You needed 
`SkLearnTfMatrixDecompositions` as an adapter. 
and 
Define a new DimensionalityReductionModel class under architectures, which inherits from BaseModel. This should have its custom set_param or get_param.
When defining SVD class, right now it is inheriting from GLM. It is totally unnecessary.

**Review by dhavala — COMMENTED** [review]
1) Sementically inheriting from GeneralzedLinearModel is not making sense. What GLM about SVD, nothing. Can you not just use the BaseModel?
What is 
2) under `transform` and `fit_transform` in the SVD implementation, why are you using np.dot. Instead you should tensorflow version.

**Pull request closed without merging**